test/PackageVersionsTest/E2EInstallerTest.php

```diff
@@ -0,0 +1,222 @@
+
+        $this->exec(__DIR__ . '/../../vendor/bin/composer global update');
+
+        $this->createArtifact();
+        file_put_contents($this->tempLocalComposerHome . '/composer.json', json_encode([
+            'name'         => 'package-versions/e2e-local',
+            'require'      => [
+                'test/package' => '2.0.0'
+            ],
+            'repositories' => [
+                [
+                    'packagist' => false,
+                ],
+                [
+                    'type' => 'artifact',
+                    'url' => $this->tempArtifact,
+                ]
+            ]
+        ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES));
+
+        $this->execInDir('composer update', $this->tempLocalComposerHome);
+        $this->assertFileNotExists(
+            $this->tempLocalComposerHome . '/vendor/ocramius/package-versions/src/PackageVersions/Versions.php'
+        );
+    }
+
+    public function testRemovingPluginDoesNotAttemptToGenerateVersions()
+    {
+        $this->createPackageVersionsArtifact();
+        $this->createArtifact();
+
+        file_put_contents($this->tempLocalComposerHome . '/composer.json', json_encode([
+           'name'         => 'package-versions/e2e-local',
+           'require'      => [
+               'test/package' => '2.0.0',
+               'ocramius/package-versions' => '1.0.0'
+           ],
+           'repositories' => [
+               [
+                   'packagist' => false,
+               ],
+               [
+                   'type' => 'artifact',
+                   'url' => $this->tempArtifact,
+               ]
+           ]
+       ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES));
+
+        $this->execInDir('composer update -vvv', $this->tempLocalComposerHome);
+        $this->assertFileExists(
+            $this->tempLocalComposerHome . '/vendor/ocramius/package-versions/src/PackageVersions/Versions.php'
+        );
+
+        $this->execInDir('composer remove ocramius/package-versions -vvv', $this->tempLocalComposerHome);
+        $this->assertFileNotExists(
+            $this->tempLocalComposerHome . '/vendor/ocramius/package-versions/src/PackageVersions/Versions.php'
+        );
+    }
+
+    private function createPackageVersionsArtifact()
+    {
+        $zip = new ZipArchive();
+
+        $zip->open($this->tempArtifact . '/ocramius-package-versions-1.0.0.zip', ZipArchive::CREATE);
+
+        $files = array_filter(
+            iterator_to_array(new RecursiveIteratorIterator(
+                new RecursiveCallbackFilterIterator(
+                    new RecursiveDirectoryIterator(realpath(__DIR__ . '/../../'), RecursiveDirectoryIterator::SKIP_DOTS),
+                    function (SplFileInfo $file) {
+                        $filePath = substr($file->getRealPath(), strlen(realpath(__DIR__ . '/../../')) + 1);
+
+                        if (substr($filePath, 0, 4) === '.git'
+                            || substr($filePath, 0, 5) === '.idea'
```

Comment:
You got a link? I couldn't find anything other than `SKIP_DOTS` which doesn't consider anything other than `..` & `.`